src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,210 @@
+            tuple corresponds to the lower and upper bounds of the optimal iteration range.
+        iter_lowupp_factor: Lower and upper multiplication factors. The first and second
+            elements of the tuple corresponds to the lower and upper multiplication
+            factors, respectively. We require the lower multiplication factor to be
+            strictly greater than one, whereas the upper multiplication factor is
+            required to be strictly less than one.
+        recomp_factor: Failed-to-converge recomputation factor. Factor by which the
+            time step will be multiplied in case the solution must be recomputed (see
+            documentation of `next_time_step` method).
+        recomp_max: Failed-to-converge maximum recomputation attempts. The maximum
+            allowable number of consecutive recomputation attempts. If `recomp_max` is
+            exceeded, an error will be raised.
+        print_info. Print time-stepping information.
+
+    Example:
+        # The following is an example on how to construct a time-stepping object
+        tsc = pp.TimeSteppingControl(
+            schedule=[0, 10],
+            dt_init=0.5,
+            dt_min_max=(0.1, 2),
+            iter_max=10,
+            iter_optimal_range=(4, 7),
+            iter_lowupp_factor=(1.1, 0.9),
+            recomp_factor=0.5,
+            recomp_max=5,
+            print_info=True
+        )
+        # To inspect the current attributes of the object
+        print(tsc)
+
+    Attributes:
+        dt (float): Time step.
+        dt_init (float): Intial time step.
+        dt_max (float): Maximum time step.
+        dt_min (float): Minimum time step.
+        is_constant (bool): Constant time step flag.
+        iter_low (int): Lower bound of the optimal iteration range.
+        iter_low_factor (float): Lower multiplication factor. Strictly greater than one.
+        iter_max (int): Maximum number of iterations.
+        iter_upp (int): Upper bound of the optimal iteration range.
+        iter_upp_factor (float): Upper multiplication factor. Strictly lower than one.
+        recomp_factor (float): Recomputation factor. Strictly lower than one.
+        recomp_max (int): Maximum number of recomputation attempts.
+        schedule (list): List of scheduled times including initial and final times.
+        time (float): Current time.
+        time_final (float): Final simulation time.
+        time_init (float): Initial simulation time.
+    """
 
     def __init__(
         self,
         schedule: list,
         dt_init: float,
-        dt_min_max: Tuple[float, float],
+        constant_dt: bool = False,
+        dt_min_max: Optional[tuple[float, float]] = None,
         iter_max: int = 15,
-        iter_optimal_range: Tuple[int, int] = (4, 7),
-        iter_lowupp_factor: Tuple[float, float] = (1.3, 0.7),
+        iter_optimal_range: tuple[int, int] = (4, 7),
+        iter_lowupp_factor: tuple[float, float] = (1.3, 0.7),
```

Comment:
This could be a matter of taste, but some defaults in academic reservoir simulators are:
* A factor of  0.5 for decreasing the time step
* A factor of 1.25 for increasing the time step
* For constant time step just setting up `iter_lowupp_factor = (1.0,1.0)` and delete the variable constant_dt.
This can be found in "Principles of Applied Reservoir Simulation" John R. Fnachi.

Reply:
This is surely a matter of taste. 

As for the constant, a compromise could be to keep constant_dt as a user input, but implement it as setting both relaxation factors to 1. What do you think @jhabriel?

Reply:
Thanks Omar. I did not know about the standards in reservoir simulation (since I primarily worked with unsaturated systems). The default parameters can be easily changed upon consensus, and I don't have strong opinions about them. 

As for the constant time step being implemented using unitary relaxation factors, I do have strong opinions on why this is not a good idea.

Bypassing the algorithm and always returning the same time step is (1) cleaner, (2) arguably faster, (3) easier to debug, and (4) more scalable. One then may argue: Why incorporate constant dt into the adaptive time step strategy? As a matter of fact, the only reason is the possibility to access the schedule, which I think is really important.

Another way to see this is: users that are only interested in linear transient models should not be required to read through the whole documentation when debugging their models, i.e.: to check what are these relaxation factors. 

As for scalability, in the future, we might want to incorporate other time-stepping control algorithms, this will most likely require creating an abstract time control class. One can then think of the constant time step class as the most basic type of time control, and keeping implementations separated will easy future implementations.